cognite/neat/graph/extractors/_base.py

```diff
@@ -0,0 +1,12 @@
+from abc import abstractmethod
+from collections.abc import Iterable
+
+
+class BaseExtractor:
+    """This is the base class for all extractors. It defines the interface that
+    extractors must implement.
+    """
+
+    @abstractmethod
+    def extract(self) -> Iterable[tuple]:
```

Comment:
Extract should always return `Iterable[tuple[URIRef, URIRef, URIRef|Literal]]`. Checkout issue:
https://github.com/cognitedata/neat/issues/213